### PR TITLE
Upgrade mysql to 8.0.40

### PR DESCRIPTION
**Pull request by realsdx**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Fixes 40 CVEs --
CVE-2024-21193
CVE-2024-21194
CVE-2024-21162
CVE-2024-21157
CVE-2024-21130
CVE-2024-20996
CVE-2024-21129
CVE-2024-21159
CVE-2024-21135
CVE-2024-21173
CVE-2024-21160
CVE-2024-21125
CVE-2024-21134
CVE-2024-21127
CVE-2024-21142
CVE-2024-21166
CVE-2024-21163
CVE-2024-21203
CVE-2024-21219
CVE-2024-21247
CVE-2024-21237
CVE-2024-21231
CVE-2024-21213
CVE-2024-21218
CVE-2024-21197
CVE-2024-21230
CVE-2024-21207
CVE-2024-21201
CVE-2024-21198
CVE-2024-21238
CVE-2024-21196
CVE-2024-21239
CVE-2024-21199
CVE-2024-21241
CVE-2024-21236
CVE-2024-21212
CVE-2024-21096
CVE-2024-21171
CVE-2024-21165
CVE-2023-46219

###### Change Log  

- Upgrade to 8.0.40

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=659397&view=ms.vss-test-web.build-test-results-tab&runId=265508&resultId=100000&paneView=attachments (25 known failures in ptests)
- Local Build & install

### CODE REVIEW

**Review by xordux — DISMISSED** [review]

**Comment by realsdx** [comment]
Retargeted to fasttrack/2.0

File: SPECS/mysql/mysql.spec [hunk 1]
```diff
@@ -98,6 +97,10 @@ fi
 %{_libdir}/pkgconfig/mysqlclient.pc
 
 %changelog
+* Fri Oct 18 2024 Sudipta Pandit <sudpandit@microsoft.com> - 8.0.37-1
```
Comment (anphel31) [inline SPECS/mysql/mysql.spec:100]: This should be version 8.0.40-1

File: SPECS/mysql/mysql.spec [hunk 2]
```diff
@@ -98,6 +97,10 @@ fi
...
+- Upgrade to 8.0.40 to fix CVE-2024-21096
```
Comment (anphel31) [inline SPECS/mysql/mysql.spec:101]: Can you list the other CVEs fixed by this upgrade? Or mention that it fixes 19 CVES

File: SPECS/mysql/mysql.spec (hunk 1 above)
Comment (jslobodzian) [inline SPECS/mysql/mysql.spec:100]: version number is wrong in changelog

Comment (realsdx) [reply SPECS/mysql/mysql.spec]: Fixed

File: SPECS/mysql/mysql.spec (hunk 2 above)
Comment (anphel31) [reply SPECS/mysql/mysql.spec]: It would be best to list all of the CVEs that we know are fixed

**Comment by realsdx** [comment]
New buddy build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=660991&view=results

**Review by anphel31 — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**